backends/dpdk/dpdkHelpers.cpp

```diff
@@ -346,6 +437,166 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
     return false;
 }
 
+/* This function processes the hash parameters and stores it in
+   "components" which is further used for generating hash instruction
+*/
+void ConvertStatementToDpdk::processHashParams(const IR::Argument* field,
+                                               IR::Vector<IR::Expression>& components) {
+    if (auto params = field->expression->to<IR::Member>()) {
+        auto typeInfo = typemap->getType(params);
+        if (auto typeheader = typeInfo->to<IR::Type_Header>()) {
+            for (auto it : typeheader->fields) {
+                IR::ID name(params->member.name + "." + it->name);
+                auto m = new IR::Member(new IR::PathExpression(IR::ID("h")), name);
+                components.push_back(m);
+            }
+        } else {
+            components.push_back(params);
+        }
+    } else if (auto s = field->expression->to<IR::StructExpression>()) {
+        for (auto field1 : s->components) {
+            if (auto params = field1->expression->to<IR::Member>()) {
+                auto typeInfo = typemap->getType(params);
+                if (auto typeheader = typeInfo->to<IR::Type_Header>()) {
+                    for (auto it : typeheader->fields) {
```

Comment:
Maybe this should be a utility function?

Reply:
Since, the computation inside 'for loop' done is specific to 'Hash' extern processing. So, this has not been written as utility function.

Reply:
Yes, but you are using it at least twice